src/core/comm/coll.cc

```diff
@@ -270,16 +264,9 @@ int collInit(int argc, char* argv[])
   CHECK_MPI(MPI_Comm_get_attr(MPI_COMM_WORLD, MPI_TAG_UB, &tag_ub, &flag));
   assert(flag);
   mpi_tag_ub = *tag_ub;
-  // create mpi comms
-  mpi_comms.resize(MAX_NB_COMMS, MPI_COMM_NULL);
-  for (int i = 0; i < MAX_NB_COMMS; i++) { CHECK_MPI(MPI_Comm_dup(MPI_COMM_WORLD, &mpi_comms[i])); }
+  mpi_comms.clear();
```

Comment:
I don't see why this is necessary. is this because you're unsure that the statically declared vector is not empty?

Reply:
Yes, I am not sure if the vector will be empty by default. Maybe I can do a check instead of using the clear.

Reply:
fixed.